data-pipeline.yml

```diff
@@ -0,0 +1,229 @@
+version: 1
+tasks:
+  - ID: commit-retrieval
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '12 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc # TODO FIX ME BEFORE MERGING
+    payload:
+      maxRunTime: 43200
+      image: mozilla/bugbug-commit-retrieval
+
+      artifacts:
+        public/commits.json.xz:
+          path: /data/commits.json.xz
+          type: file
+
+    routes:
+      # - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
+      - notify.irc-channel.#bugbug.on-failed
+    metadata:
+      name: bugbug commit retrieval
+      description: bugbug commit retrieval
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: bugs-retrieval
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '24 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc
+    payload:
+      env:
+        TC_SECRET_ID: project/relman/bugbug/production
+      maxRunTime: 86400
+      image: mozilla/bugbug-bugs-retrieval
+
+      artifacts:
+        public/bugs.json.xz:
+          path: /data/bugs.json.xz
+          type: file
+
+      features:
+        taskclusterProxy:
+          true
+    scopes:
+      - "secrets:get:project/relman/bugbug/production"
+
+    routes:
+      # - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
+      - notify.irc-channel.#bugbug.on-failed
+    metadata:
+      name: bugbug bugs retrieval
+      description: bugbug bugs retrieval
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: train-component
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '7 hour'}
```

Comment:
I think the deadline should be the maximum runtime of the task + the maximum runtime of the tasks it depends on.